File: src/MediCore.API/MediCore.Domain/Entities/Financial/Invoice.cs
using MediCore.Domain.Common;
using MediCore.Domain.Enums;

namespace MediCore.Domain.Entities.Financial;

public class Invoice : BaseEntity
{
    public Guid TenantId { get; set; }
    public Guid BranchId { get; set; }
    public Guid PatientId { get; set; }
    public Guid? EncounterId { get; set; }
    public Guid? DoctorId { get; set; }
    public string InvoiceNumber { get; set; } = string.Empty;
    public DateOnly InvoiceDate { get; set; }
    public decimal Subtotal { get; set; }
    public decimal DiscountAmount { get; set; } = 0;
    public string? DiscountReason { get; set; }
    public decimal TaxAmount { get; set; } = 0;
    public decimal TotalAmount { get; set; }
    public decimal PaidAmount { get; set; } = 0;
    public decimal RemainingAmount { get; set; }
    public decimal InsuranceCoverage { get; set; } = 0;
    public InvoiceStatus Status { get; set; } = InvoiceStatus.Draft;
    public string? Notes { get; set; }

    // Navigation Properties
    public ICollection<InvoiceItem> Items { get; set; } = new List<InvoiceItem>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
}

File: src/MediCore.Desktop/lib/features/invoices/domain/entities/invoice.dart
import 'invoice_item.dart';

class Invoice {
  final String id;
  final String patientId;
  final String? encounterId;
  final String invoiceNumber;
  final DateTime invoiceDate;
  final double subtotal;
  final double discountAmount;
  final double taxAmount;
  final double totalAmount;
  final double paidAmount;
  final double remainingAmount;
  final String status;
  final List<InvoiceItem> items;
  final DateTime createdAt;

  const Invoice({
    required this.id,
    required this.patientId,
    this.encounterId,
    required this.invoiceNumber,
    required this.invoiceDate,
    required this.subtotal,
    required this.discountAmount,
    required this.taxAmount,
    required this.totalAmount,
    required this.paidAmount,
    required this.remainingAmount,
    required this.status,
    required this.items,
    required this.createdAt,
  });
}
